### PR TITLE
Fix pagination mocking

### PR DESCRIPTION
**Pull request by mottalrd**
The value TOTAL is the total number of results available (not the total number of results returned).

### CODE REVIEW

File: lib/ht/search_client/test.rb [hunk 1]
```diff
@@ -38,11 +38,11 @@ def stay_class?
       self.class.to_s =~ /Stay/
     end
 
-    def mocked_results(properties)
+    def mocked_results(properties, total)
       {
         page:     params.fetch(:page),
         per_page: params.fetch(:per_page),
-        total:    properties.length,
+        total:    total,
```
Comment (andycroll) [inline lib/ht/search_client/test.rb]: could use the original as the default?

Comment (mottalrd) [inline lib/ht/search_client/test.rb]: it's tricky. The original is correct only when `total < per_page`
I can add this logic and warning the user....

Comment (pedrocunha) [inline lib/ht/search_client/test.rb]: How about:
- by default total should be `[per_page, properties.length].min`
- accepts custom param

Comment (mottalrd) [inline lib/ht/search_client/test.rb]: > by default total should be [per_page, properties.length].min

wouldn't that lead to this incorrect scenario?

```
properties: [ put 20 properties here ]
page: 2
per_page: 32
total: 20
```

here is the `page` that is breaking the equation

Comment (pedrocunha) [inline lib/ht/search_client/test.rb]: Remember that the idea is *"mocking"* not validating things - I agree that before there was no control over the `total` which might be interesting to test pagination in a view for example hence be able to control that value. That's why my suggestion: by default try to be nice, if you supply something then you have to guarantee all is good.  

If you really want to add validation then I think we should raise a error if you try to do the above.

My 2 cents.

Comment (mottalrd) [inline lib/ht/search_client/test.rb]: Then I would say that I simply default to `properties.size` like it was before.
Everything else is left to the caller.
I pushed the change

**Comment by andycroll** [comment]
:+1:

**Pull request merged**